### DISCUSSION TITLE
How to show columns of DataGrid when the binding ItemsSource has zero element?

### DISCUSSION THREAD

**Discussion by GF-Huang** [HandyOrg/HandyControl#723] [Q&A] [answered]
When the ItemsSource have no element, the columns of DataGrid not display.

**Comment by ghost1372** [comment]
Use `hc:Empty.ShowEmpty="true"` AttachedProperty

https://hosseini.ninja/handycontrol/extend_controls/empty/

**Reply by GF-Huang** [reply]
I means keep the columns header display even if no data.

**Reply by GF-Huang** [reply]
I found this thing hide `DataGrid`:

https://github.com/HandyOrg/HandyControl/blob/9d9b7b9751b2690d1ba35ca344b1eb1e2d9efb36/src/Shared/HandyControl_Shared/Themes/Styles/DataGrid.xaml#L294

Any good ideas to change this behavior without re-write a control template?

**Reply by qwfy168** [reply]
你找到方法了吗，没有表头好不习惯

**Reply by GF-Huang** [reply]
后来我发现用一个图案下面加个文字表示没有数据，效果要比显示个有表头的空表要美观。

**Comment by NaBian** [comment, marked as answer]
fixed in 77f273f6aecc9b5db82f68307ea3111e48b476ea
